src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3218,12 +3247,78 @@ public ConstructorEdit(INamedTypeSymbol oldType)
             }
         }
 
+        private static bool TryAddParameterInsertOrDeleteEdits(ArrayBuilder<SemanticEditInfo> semanticEdits, ISymbol containingSymbol, SemanticModel? otherModel, EditAndContinueCapabilitiesGrantor capabilities, Func<SyntaxNode, SyntaxNode?>? syntaxMap, EditScript<SyntaxNode> editScript, CancellationToken cancellationToken, out bool notSupportedByRuntime)
+        {
+            notSupportedByRuntime = false;
+
+            // Since we're inserting (or deleting) a parameter node, oldSymbol (or newSymbol) would have been null,
+            // and a symbolkey won't map to the other compilation because the parameters are different, so we have to go back to the edit map
+            // to find the declaration that contains the parameter, and its partner
+
+            // Some sanity checks
+            if (otherModel is null ||
+                containingSymbol.DeclaringSyntaxReferences.Length != 1)
+            {
+                return false;
+            }
+
+            // We don't support delegate parameters
+            if (containingSymbol.ContainingType.IsDelegateType())
+            {
+                return false;
+            }
+
+            // Find the node that matches this declaration
+            SyntaxNode otherContainingNode;
+            var containingNode = containingSymbol.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken);
+            if (editScript.Match.TryGetOldNode(containingNode, out var oldNode))
+            {
+                otherContainingNode = oldNode;
+            }
+            else if (editScript.Match.TryGetNewNode(containingNode, out var newNode))
+            {
+                otherContainingNode = newNode;
+            }
+            else
+            {
+                return false;
+            }
+
+            var otherContainingSymbol = otherModel.GetDeclaredSymbol(otherContainingNode, cancellationToken);
+            if (otherContainingSymbol is null || !AllowsDeletion(otherContainingSymbol))
+            {
+                return false;
+            }
+
+            // Now we can work out which is the old and which is the new, depending on which map we found
+            // the match in
+            var oldSymbol = (otherContainingNode == oldNode) ? otherContainingSymbol : containingSymbol;
+            var newSymbol = (otherContainingNode == oldNode) ? containingSymbol : otherContainingSymbol;
+
+            if (CanAddNewMember(oldSymbol, capabilities))
```

Comment:
>CanAddNewMember [](http://example.com/codeflow?start=16&length=15)

Nit: consider inverting this condition to follow the pattern of if statements above.